PKCS115_Cipher.decrypt - broken contract since 3.10.3 - returns bytearray instead bytes

Hello
Since this change: https://github.com/Legrandin/pycryptodome/commit/853054937e29e6cd3dc61228be1dc768b688460b#
that was done due to https://github.com/Legrandin/pycryptodome/issues/481
the PKCS115_Cipher.decrypt returns bytearray instead of bytes.
The documentation clearly states that byte string (bytes) is to be returned
https://github.com/Legrandin/pycryptodome/blame/master/lib/Crypto/Cipher/PKCS1_v1_5.py#L142
```        
Returns (byte string):
            It is either the original message or the ``sentinel`` (in case of an error).
```
also the typing specification states bytes: https://github.com/Legrandin/pycryptodome/blob/master/lib/Crypto/Cipher/PKCS1_v1_5.pyi#L14
```
    def decrypt(self, ciphertext: Buffer,
                sentinel: Any,
                expected_pt_len: Optional[int]) -> bytes: ...
```
and yet since the change the bytearray is used

Simple test to replicate (tested on Python 3.7.5)
```
from unittest import TestCase
from Cryptodome.PublicKey import RSA
from Cryptodome.Cipher import PKCS1_v1_5


class Test_3_10_3_BreakingChange(TestCase):
    def setUp(self):
        key = RSA.generate(2048)
        private_key_data = key.export_key()
        public_key_data = key.publickey().export_key()
        public_key = RSA.import_key(public_key_data)
        private_key = RSA.import_key(private_key_data)

        self.data = b'SomeDataToEncrypt'
        self.encrypt_rsa = PKCS1_v1_5.new(public_key)
        self.decrypt_rsa = PKCS1_v1_5.new(private_key)

    def test_decrypt(self):
        enc_data = self.encrypt_rsa.encrypt(self.data)
        assert isinstance(enc_data, bytes)
        dec_data = self.decrypt_rsa.decrypt(ciphertext=enc_data, sentinel=b'justTest')
        assert isinstance(dec_data, bytes)
```
For:
```
pip freeze |grep cryptodomex
pycryptodomex==3.10.3
```
```
pip freeze |grep cryptodomex
pycryptodomex==3.10.4
```
Result:
```
test_PKCS1_v1_5_decrypt.py::Test_3_10_3_BreakingChange::test_decrypt FAILED [100%]
test/unit_tests/test_PKCS1_v1_5_decrypt.py:17 (Test_3_10_3_BreakingChange.test_decrypt)
self = <test_PKCS1_v1_5_decrypt.Test_3_10_3_BreakingChange testMethod=test_decrypt>

    def test_decrypt(self):
        enc_data = self.encrypt_rsa.encrypt(self.data)
        assert isinstance(enc_data, bytes)
        dec_data = self.decrypt_rsa.decrypt(ciphertext=enc_data, sentinel=b'justTest')
>       assert isinstance(dec_data, bytes)
E       AssertionError: assert False
E        +  where False = isinstance(bytearray(b'SomeDataToEncrypt'), bytes)

test_PKCS1_v1_5_decrypt.py:22: AssertionError
```

For:
```
pip freeze |grep cryptodomex
pycryptodomex==3.10.1
```
Result:
```
test_PKCS1_v1_5_decrypt.py::Test_3_10_3_BreakingChange::test_decrypt 

============================== 1 passed in 1.93s ===============================

Process finished with exit code 0
PASSED [100%]
```

--- a/lib/Crypto/Cipher/PKCS1_v1_5.py
+++ b/lib/Crypto/Cipher/PKCS1_v1_5.py
@@ -163,7 +163,7 @@ class PKCS115_Cipher:
             as possible even if ``sentinel`` is returned (i.e. in case of
             incorrect decryption).
 
-            .. __: http://www.bell-labs.com/user/bleichen/papers/pkcs.ps
+            .. __: https://dx.doi.org/10.1007/BFb0055716
         """
 
         # See 7.2.2 in RFC8017

--- a/lib/Crypto/Cipher/PKCS1_v1_5.py
+++ b/lib/Crypto/Cipher/PKCS1_v1_5.py
@@ -183,7 +183,7 @@ class PKCS115_Cipher:
         em = long_to_bytes(m_int, k)
 
         # Step 3 (not constant time when the sentinel is not a byte string)
-        output = bytearray(k)
+        output = bytes(bytearray(k))
         if not is_bytes(sentinel) or len(sentinel) > k:
             size = _pkcs1_decode(em, b'', expected_pt_len, output)
             if size < 0:

